### DISCUSSION TITLE
Scroll bar with command execution indicator

### DISCUSSION THREAD

**Discussion by SenseiMarv** [ghostty-org/ghostty#3870] [Feature Requests, Ideas]
In iTerm2, when shell integration is enabled, there is a scroll bar on the right that shows the lines where a command has been executed:

<img width="573" alt="image" src="https://github.com/user-attachments/assets/4550049a-ac2b-4b54-b444-d31326b42dda" />

The indicator makes it easier to navigate the history by giving a visual clue where the scrollbar can be clicked to jump to a command execution. I would like to discuss if something similar can be implemented for Ghostty to improve the navigability of the history. This would also go hand in hand with the sticky scroll suggestion: https://github.com/ghostty-org/ghostty/discussions/3603

**Comment by pluiedev** [comment]
We don't have a scrollbar in general (#111), though this is an intriguing idea.

**Comment by mixmastamyk** [comment]
That's also used for search results, similar to how browsers do it.  Would love that as well.
